zcash_primitives/src/note_encryption.rs

```diff
@@ -238,6 +304,20 @@ mod tests {
             let note = to.create_note(tv.v, rcm, &JUBJUB).unwrap();
             assert_eq!(note.cm(&JUBJUB), cmu);
 
+            //
```

Comment:
Add additional tests for failure to decrypt for various reasons:

1. invalid diversifier
2. invalid tag
3. invalid commitment
4. slice length too small?